packages/ra-ui-materialui/src/button/CreateButton.tsx

```diff
@@ -41,50 +42,56 @@ const CreateButton = (props: CreateButtonProps) => {
     const isSmall = useMediaQuery((theme: Theme) =>
         theme.breakpoints.down('md')
     );
+    const state = merge(scrollStates[String(scrollToTop)], initialState);
```

Comment:
The way to access values inside a `Map` is using `get()`

```suggestion
    const state = merge(scrollStates.get(String(scrollToTop)), initialState);
```